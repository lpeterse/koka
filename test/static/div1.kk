type nat {
  Zero
  Succ( :nat)
}


fun h(x,y) {
  match(x) {
    Zero -> match(y) {
             Zero -> Zero
             Succ(s) -> h(x,s)
            }
    Succ(s) -> h(s,y)
  }
}

