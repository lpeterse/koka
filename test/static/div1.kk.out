static/div1/nat: V
Succ : (nat) -> nat
Zero : nat
h : (nat, nat) -> nat
isSucc : (nat : nat) -> bool
isZero : (nat : nat) -> bool